src/components/shapeStackUtils.js

```diff
@@ -0,0 +1,54 @@
+import { id as poolId } from './nodes/pool';
+import { id as laneId } from './nodes/poolLane';
+
+export default function ensureShapeIsNotCovered(shape) {
+  if (isPool(shape)) {
+    bringPoolToFront(shape);
+  }
+
+  const parentPool = getElementPool(shape);
+
+  if (parentPool) {
+    bringPoolToFront(parentPool);
+  }
+
+  if (isNotLane(shape) && !isPool(shape)) {
+    bringShapeToFront(shape);
+  }
+}
+
+function isNotLane(shape) {
+  return shape.component.node.type !== laneId;
+}
+
+function bringPoolToFront(poolShape) {
+  bringShapeToFront(poolShape);
+
+  const hasLanes = poolShape.component.node.definition.processRef.laneSets[0];
+  if (!hasLanes) {
+    return;
+  }
+
+  const poolLanes = poolShape
+    .getEmbeddedCells()
+    .filter(cell => cell.component && cell.component.node.type === laneId);
+  const poolLaneElements = poolShape
+    .getEmbeddedCells()
+    .filter(cell => cell.component && cell.component.node.type !== laneId);
+
+  poolLanes.forEach(bringShapeToFront);
+  poolLaneElements.forEach(bringShapeToFront);
+}
+
+function bringShapeToFront(shape) {
+  shape.toFront({ deep: true });
+}
+
+function getElementPool(shape) {
+  return shape.component.node.pool;
+}
+
+function isPool(shape) {
+  return shape.component.node.type === poolId;
+}
+
```

Comment:
I thought about adding unit tests for this, but I wasn't sure it would add any value. The heavy-lifting is accomplished by calls to `toFront`, and tests would essentially be verifying that `toFront` was called in the correct order, the correct number of times.

There might be better combinations of calling `toFront` that also fix overlapping issues, or maybe there's a different JointJS function we could use. I think unit tests could slow us down in those cases, and I think this functionality is sufficiently covered by our e2e tests. 

Maybe that's a hint that this isn't the best (or correct abstraction) right now, but I think it's still much better than having all this logic spread into `Modeler.vue`.